File: csharp/gmsec5/dox/field/BinaryField.cs
/*
 * Copyright 2007-2024 United States Government as represented by the
 * Administrator of The National Aeronautics and Space Administration.
 * No copyright is claimed in the United States under Title 17, U.S. Code.
 * All Rights Reserved.
 */

/* This file is used for generating Doxygen documentation; it only contains
   stubs of constructors and methods associated with the class.
*/


namespace GMSEC.API5 {

/// <summary>
/// Specialized Field class that can be used to store a raw data value.
/// </summary>

public class BinaryField : Field {

  /// <summary>Creates a BinaryField object with the given data</summary>
  ///
  /// <param name="name">The name of the field</param>
  /// <param name="data">Array of raw data</param>
  /// <param name="isHeader">Use to indicate if the field is a header field (default is false)</param>
  ///
  /// <exception cref="GmsecException">Thrown if the name is null, is an empty string, or otherwise if the field name is non-compliant.</exception>
  /// <exception cref="GmsecException">Thrown if the data array is null</exception>
  public BinaryField(string name, byte[] data, bool isHeader = false) {
  }

  /// <summary>Copy constructor.</summary>
  ///
  /// <param name="other">The other BinaryField object to copy</param>
  public BinaryField(BinaryField other) {
  }

  /// <summary>Returns the length of the data associated with the field</summary>
  ///
  /// <returns>The length of the data associated with the field</returns>
  public int GetLength() {
  }

  /// <summary>Returns the data associated with the field.</summary>
  ///
  /// <returns>The data associated with the field.</returns>
  public byte[] GetValue() {
}

}

}

File: nodejs/gmsec5/dox/BinaryField.java
/*
 * Copyright 2007-2025 United States Government as represented by the
 * Administrator of The National Aeronautics and Space Administration.
 * No copyright is claimed in the United States under Title 17, U.S. Code.
 * All Rights Reserved.
 */


/**
 * @file BinaryField.java
 */

package gmsec;


/**
 * Specialized Field class that can be used to retain raw data.
 */
public class BinaryField extends Field
{
	/**
	 * Constructor.
	 *
	 * @param name Name of the field.
	 * @param data A BlobArray containing raw data to associate with the field.
	 *
	 * @throws SWIG_Exception Thrown if the field name string is null.
	 * @throws SWIG_Exception Thrown if the BlobArray is null.
	 * @throws GmsecException Thrown if the field name string is an empty string or is otherwise non-compliant.
	 */
	public BinaryField(name, data);


	/**
	 * Constructor.
	 *
	 * @param name      Name of the field.
	 * @param data      A BlobArray containing raw data to associate with the field.
	 * @param is_header Used to indicate if Field is a header field.
	 *
	 * @throws SWIG_Exception Thrown if the field name string is null.
	 * @throws SWIG_Exception Thrown if the BlobArray is null.
	 * @throws GmsecException Thrown if the field name string is an empty string or is otherwise non-compliant.
	 */
	public BinaryField(name, data, is_header);


	/**
	 * Copy constructor.
	 *
	 * @param other The other BinaryField object to copy.
	 *
	 * @throws SWIG_Exception Thrown if the given BinaryField object is null.
	 */
	public BinaryField(other);


	/**
	 * Returns the data stored within the BinaryField.
	 *
	 * @return A BlobArray object.
	 */
	public getValue();
}
